### PR TITLE
Fixed number of visits on module statsbestcustomers

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Fixed number of visits on module statsbestcustomers
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes PrestaShop/PrestaShop#22943
| How to test?      | Cf PrestaShop/PrestaShop#22943 & #16 (from @jf-viguier)

### CODE REVIEW

File: statsbestcustomers.php [hunk 1]
```diff
@@ -167,12 +171,12 @@ public function getData()
...
 			), 0) as totalValidOrders
 		FROM `' . _DB_PREFIX_ . 'customer` c
-		INNER JOIN `' . _DB_PREFIX_ . 'orders` o ON o.id_customer = c.id_customer
 		LEFT JOIN `' . _DB_PREFIX_ . 'guest` g ON c.`id_customer` = g.`id_customer`
 		LEFT JOIN `' . _DB_PREFIX_ . 'connections` co ON g.`id_guest` = co.`id_guest` AND co.date_add BETWEEN ' . $this->getDate() . '
-		WHERE o.date_add BETWEEN ' . $this->getDate()
-            . Shop::addSqlRestriction(Shop::SHARE_CUSTOMER, 'c') .
-            ' GROUP BY c.`id_customer`, c.`lastname`, c.`firstname`, c.`email`';
+		WHERE co.date_add BETWEEN ' . $this->getDate() . '
+		AND c.id_customer IN (SELECT o.id_customer FROM `' . _DB_PREFIX_ . 'orders` o)
```
Comment (PierreRambaud) [inline statsbestcustomers.php:177]: Why using a subquery when you can use an INNER join? :thinking:

Comment (Progi1984) [reply statsbestcustomers.php:177]: INNER JOIN duplicate lines for each order, so the count of total connections are invalid.

Comment (PierreRambaud) [reply statsbestcustomers.php:177]: You can also use distinct :thinking: 
I'm afraid of performance problem when you have thousands of orders..

**Comment by florine2623** [comment]
Hello @Progi1984 ,

Thanks for the PR, LGTM !

**Comment by Progi1984** [comment]
Thanks @florine2623

**Pull request merged**